pkg/k8s/network_policy.go

```diff
@@ -52,26 +51,14 @@ func GetPolicyLabelsv1(np *networkingv1.NetworkPolicy) labels.LabelArray {
 }
 
 func parseNetworkPolicyPeer(namespace string, peer *networkingv1.NetworkPolicyPeer) *api.EndpointSelector {
-
-	// TODO add unit test (GH-3080).
 	if peer == nil {
 		return nil
 	}
 
-	var labelSelector *metav1.LabelSelector
+	var retSel *api.EndpointSelector
 
-	// Only one or the other can be set, not both
-	if peer.PodSelector != nil {
-		labelSelector = peer.PodSelector
-		if peer.PodSelector.MatchLabels == nil {
-			peer.PodSelector.MatchLabels = map[string]string{}
-		}
-		// The PodSelector should only reflect to the same namespace
-		// the policy is being stored, thus we add the namespace to
-		// the MatchLabels map.
-		peer.PodSelector.MatchLabels[k8sConst.PodNamespaceLabel] = namespace
-	} else if peer.NamespaceSelector != nil {
-		labelSelector = peer.NamespaceSelector
+	if peer.NamespaceSelector != nil {
```

Comment:
Would you be willing to use a switch here? We removed the `Only one or the other can be set, not both` comment but the intention is still roughly the same: handle one of these, preferring the namespace one. Alternatively, a comment would be great, or a comment for the whole function!